source/common/filter/tcp_proxy.cc

```diff
@@ -140,6 +140,11 @@ void TcpProxy::readDisableUpstream(bool disable) {
 
 void TcpProxy::readDisableDownstream(bool disable) {
   read_callbacks_->connection().readDisable(disable);
+  // The WsHandlerImpl class uses TCP Proxy code with a null config.
```

Comment:
Hmm OK yes I think you are right. Can you add TODO for @alyssawilk here so we make sure this works? (Or maybe it already does in the outstanding PR).